crates/file-serve/src/lib.rs

```diff
@@ -100,17 +105,45 @@ impl Server {
         self.addr.as_str()
     }
 
+    /// Whether the server is running
+    pub fn is_running(&self) -> bool {
+        self.running.load(Ordering::Acquire)
+    }
+
     /// Start the webserver
-    pub fn serve(&self) -> Result<std::convert::Infallible, Error> {
+    pub fn serve(&self) -> Result<(), Error> {
```

Comment:
The commit title has `!` but can you add a `BREAKING CHANGE:` section that summarizes what part of the API this breaks?